src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodCodeNode.cs

```diff
@@ -30,6 +30,22 @@ public MethodCodeNode(MethodDesc method)
             _method = method;
         }
 
+        /// <summary>
+        /// Validates that it will be possible 
+        /// </summary>
+        /// <param name="method"></param>
+        public static void CheckCanGenerateMethod(MethodDesc method)
+        {
+            MethodSignature signature = method.Signature;
+
+            EETypeNode.CheckCanGenerateEEType(signature.ReturnType);
```

Comment:
This is a pretty expensive check. I think what we actually need is somewhere along the lines of "make sure `Get_CORINFO_SIG_INFO` can succeed".

Maybe this method should also be part of `CompilationModuleGroup` or something like that. This file is RyuJIT specific. (We should really make the split of these into separate assemblies so that it gets more obvious.)